### PR TITLE
Refactors cinematics.

### LINKED ISSUE #21594: Round end breaks if the nuke detonates not on station
No score screen is shown, and the animation for the syndies failing to detonate on station doesn't play.
It just says station destroyed by nuclear device, rebooting in 45 seconds.

### LINKED ISSUE #23189: Post-nuke leaves screen too big
image

You can see that the view range is bigger than the HUD.

### LINKED ISSUE #23609: Nuking a Space Hotel level will result in a Syndicate Major Victory
...despite the fact that they're not nuking the station.

Makes me wish that nukes detonated off the main station Z-level didn't end the round.

### LINKED ISSUE #25522: If the nuke goes off on a non-nuke ops round, the end splash screen and cross server message say it was a self destruct
The reason for the nuke exploding seems to be tied to the mode, rather than the type of nuke that was set off.

### LINKED ISSUE #25964: Nuke detonated at the syndicate base states that the Station has been nuked and plays the station nuke cutscene
Guy detonated nuke at syndicate base said station was destroyed and played the normal station nuked message

### LINKED ISSUE #30869: Nukeops isn't roundending properly.
I've seen this issue about in game before and basically when nuke triggers SUCESSFULLY, IN THE STATION, it decides to give no animation of station blowing up, just static screen of station being fine and once the static screen disappears, round ends but there is NO INFO, nothing about nuke ops, their members, their equipment and other stuff indicating round end apart from credits I believe?

@AnturK i swear you told me on irc that you were going to investigate this

https://atlantaned.space/newSS13tools/round.php?round=76152 this has been happening for a while now but can't remember how long.

### PR DESCRIPTION
**Pull request by AnturK**
Fixes #25964 
Fixes #30869 
Fixes #25522
Fixes #23609
Fixes #23189
Fixes #21594
Probably more but i don't feel like looking more.

Warning. Highly toxic gamemode code inside.

### CODE REVIEW

**Comment by DoctoreMad** [comment]
Did you know? There is a round end image for the syndicate nuking their own base in the code.

**Comment by XDTM** [comment]
i wonder if i could work a cinematic into a hallucination

File: code/datums/cinematic.dm [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+	var/datum/callback/special_callback //For special effects synced with animation (explosions after the countdown etc)
+	var/cleanup_time = 300 //How long for the final screen to remain
+
+/datum/cinematic/New()
+	GLOB.cinematics += src
+	screen = new(src)
+
+/datum/cinematic/Destroy()
+	GLOB.cinematics -= src
+	qdel(screen)
```
Comment (Cyberboss) [inline code/datums/cinematic.dm]: QDEL_NULL

File: code/datums/cinematic.dm [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+			return //Can't play two global or local cinematics at the same time
+
+	//Close all open windows if global
+	if(is_global)
+		for (var/datum/html_interface/hi in GLOB.html_interfaces)
+			hi.closeAll()
+		SStgui.close_all_uis()
+
+
+	for(var/mob/M in GLOB.mob_list)
```
Comment (Cyberboss) [inline code/datums/cinematic.dm]: us player_list instead

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/cinematic.dm (hunk 2 above)
Comment (AnturK) [reply code/datums/cinematic.dm]: no_transform still needs to be set on all for global ones so this saves a second go over.

Comment (AnturK) [reply code/datums/cinematic.dm]: Then again, the whole no-transform is kinda debatable anyway.

**Comment by optimumtact** [comment]
Good stuff

**Pull request merged**